File: app/src/main/java/de/mm/android/longitude/database/MyDBDelegate.java
package de.mm.android.longitude.database;

import android.content.Context;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

import de.mm.android.longitude.model.ContactData;
import de.mm.android.longitude.model.ProcessEntry;

/**
 * Created by Max on 07.04.2015.
 */
public class MyDBDelegate {

    private MyDBDelegate() {
        throw new AssertionError("No Instances");
    }

    /* Insert + Update */

    public static void insertProcessEntry(Context c, ProcessEntry p){
        MyDB db = new MyDB(c);
        db.open();
        db.insertProcessEntry(p);
        db.close();
    }

    public static void mergeFriends(Context c, List<ContactData> data){
        MyDB db = new MyDB(c);
        db.open();
        db.mergeFriends(data);
        db.close();
    }

    /* Select's */

    public static ArrayList<ContactData> selectFriends(Context c) {
        MyDB db = new MyDB(c);
        db.open();
        ArrayList<ContactData> data = db.selectFriends();
        db.close();
        return data;
    }

    public static List<ProcessEntry> selectProcess(Context c, Calendar min, Calendar max){
        MyDB db = new MyDB(c);
        db.open();
        List<ProcessEntry> data = db.selectProcess(min, max);
        db.close();
        return data;
    }

    /* Delete's */

    public static void deleteProcessEntries(Context c, Calendar calendar) {
        MyDB db = new MyDB(c);
        db.open();
        db.deleteProcessEntries(calendar);
        db.close();
    }

}

File: app/src/main/java/de/mm/android/longitude/database/MyDBDelegate.kt
package de.mm.android.longitude.database

import android.content.Context

import java.util.ArrayList
import java.util.Calendar

import de.mm.android.longitude.model.ContactData
import de.mm.android.longitude.model.ProcessEntry

/**
 * Created by Max on 07.04.2015.
 */
class MyDBDelegate private constructor() {

    companion object {

        /* Insert + Update */

        fun insertProcessEntry(c: Context, p: ProcessEntry) {
            val db = MyDB(c)
            db.open()
            db.insertProcessEntry(p)
            db.close()
        }

        fun mergeFriends(c: Context, data: List<ContactData>) {
            val db = MyDB(c)
            db.open()
            db.mergeFriends(data)
            db.close()
        }

        /* Select's */

        fun selectFriends(c: Context): ArrayList<ContactData> {
            val db = MyDB(c)
            db.open()
            val data = db.selectFriends()
            db.close()
            return data
        }

        fun selectProcess(c: Context, min: Calendar?, max: Calendar?): List<ProcessEntry> {
            val db = MyDB(c)
            db.open()
            val data = db.selectProcess(min, max)
            db.close()
            return data
        }

        /* Delete's */

        fun deleteProcessEntries(c: Context, calendar: Calendar) {
            val db = MyDB(c)
            db.open()
            db.deleteProcessEntries(calendar)
            db.close()
        }
    }

}
